### PR TITLE
test on mvapich

### PR DESCRIPTION
**Pull request by simonbyrne**

### CODE REVIEW

File: .github/workflows/UnitTests.yml [hunk 1]
```diff
@@ -626,3 +626,48 @@ jobs:
       run: |
         source ${HOME}/intel/compilers_and_libraries/linux/mpi/intel64/bin/mpivars.sh release
         julia --color=yes --project -e 'using Pkg; Pkg.test()'
+
+  test-spack-mvapich:
+    timeout-minutes: 20
```
[the remark this answers is not in the document]
Comment (giordano) [reply .github/workflows/UnitTests.yml:631]: Uh, you also need the Ubuntu package `libibverbs-dev`.

[the remark this answers is not in the document]
Comment (giordano) [reply .github/workflows/UnitTests.yml:631]: With the above spack environment and the Ubuntu packages `rdma-core` and `libibverbs-dev`, I was able to build mvapich on GitHub Actions
```
==> mvapich2: Successfully installed mvapich2-2.3.7-dmysxjspupkgl5giqs4pjkuqdzpwlzpp
  Stage: 0.81s.  Autoreconf: 0.00s.  Configure: 1m 50.69s.  Build: 30m 23.42s.  Install: 22.95s.  Total: 32m 37.95s
```
It takes about 32-33 minutes just to build it (excluding the other non-external dependencies, but they're pretty quick)

[the remark this answers is not in the document]
Comment (simonbyrne) [reply .github/workflows/UnitTests.yml:631]: might be easier if you just push to this branch?

**Comment by giordano** [comment]
Small progress: I managed to build and [cache](https://github.com/JuliaParallel/MPI.jl/actions/caches) MVAPICH.  However I didn't manage to make MPIPreferences find it, but with warm cache this should finally be easier to debug.

**Comment by giordano** [comment]
Ok, this is finally working as in "MPI.jl start", but initialzing MVAPICH crashes immediately.

**Comment by giordano** [comment]
Spawn tests are failing as reported in https://github.com/JuliaParallel/MPI.jl/issues/684#issuecomment-1341937162.

**Comment by simonbyrne** [comment]
Fantastic, thanks @giordano!

**Comment by giordano** [comment]
MVAPICH tests (after excluding spawn) are now successful! Most of the other failures on julia nightly are related to https://github.com/JuliaLang/julia/issues/48039

**Comment by simonbyrne** [comment]
Thanks for your help!

**Pull request merged**